### PR TITLE
Log exceptions resulting in 5xx

### LINKED ISSUE #77: Log exceptions for 5xx by default, not just 500
```java
if (status == HttpStatus.INTERNAL_SERVER_ERROR) {
    LOG.error("Internal Server Error", throwable);
}
```

It should check the http status family/series instead.

### PR DESCRIPTION
**Pull request by whiskeysierra**
Fixes #77

### CODE REVIEW

File: src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java [hunk 1]
```diff
@@ -151,8 +151,8 @@ default void log(
             @SuppressWarnings("UnusedParameters") final Problem problem,
             @SuppressWarnings("UnusedParameters") final NativeWebRequest request,
             final HttpStatus status) {
-        if (status == HttpStatus.INTERNAL_SERVER_ERROR) {
-            LOG.error("Internal Server Error", throwable);
+        if (status.is5xxServerError()) {
```
Comment (AlexanderYastrebov) [inline src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java]: Write a test? :)

Comment (whiskeysierra) [inline src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java]: Unless I mock the static logger it's not possible.

Comment (whiskeysierra) [inline src/main/java/org/zalando/problem/spring/web/advice/AdviceTrait.java]: Maybe this http://projects.lidalia.org.uk/slf4j-test/

**Comment by AlexanderYastrebov** [comment]
:+1:

**Pull request merged**